crates/re_data_ui/src/editors.rs

```diff
@@ -257,28 +257,48 @@ fn edit_marker_shape_ui(
 
     egui::ComboBox::from_id_source("marker_shape")
         .selected_text(marker_text) // TODO(emilk): Show marker shape in the selected text
-        .width(100.0)
+        .width(ui.available_width().at_most(100.0))
         .height(320.0)
         .show_ui(ui, |ui| {
             // no spacing between list items
             ui.spacing_mut().item_spacing.y = 0.0;
 
-            // Hack needed for ListItem to click its highlight bg rect correctly:
-            ui.set_clip_rect(
-                ui.clip_rect()
-                    .with_max_x(ui.max_rect().max.x + ui.spacing().menu_margin.right),
+            let item_width = 100.0;
+
+            // workaround to force `ui.max_rect()` to reflect the content size
+            ui.allocate_space(egui::vec2(item_width, 0.0));
```

Comment:
We should try to match the outer width of the combo-box with the outer width of the dropdown, i.e. take margins into account. That's what egui does by default:

![Screenshot 2024-05-03 at 11 58 43](https://github.com/rerun-io/rerun/assets/1148717/cfa3ae8f-0383-4ed2-869e-14d713f8b763)


![Screenshot 2024-05-03 at 12 00 13](https://github.com/rerun-io/rerun/assets/1148717/2a87fae0-c3d4-4f95-95ba-973b30380958)

Reply:
Good point. Done in #6211